modules/docs/arrow-docs/docs/docs/optics/syntax/README.md

```diff
@@ -12,14 +12,10 @@ To avoid boilerplate Arrow can generate this property-like dsl using `@optics` a
 ```kotlin
 package com.example.domain
 
-@optics
-data class Street(val number: Int, val name: String)
-@optics
-data class Address(val city: String, val street: Street)
-@optics
-data class Company(val name: String, val address: Address)
-@optics
-data class Employee(val name: String, val company: Company?)
+@optics data class Street(val number: Int, val name: String)
```

Comment:
Is it mandatory to annotate each one of the implementations of the sum type (sealed class) to generate the optics for each ? Shouldn't we be able to generate optics for all the implementations on the sealed hierarchy if you just annotate the parent sealed class itself?

Reply:
Yes, it is. Otherwise, it might get complex really quickly.

Given the network example, what is generated?

```
@optics sealed class NetworkResult
data class Success(val content: String): NetworkResult()
sealed class NetworkError: NetworkResult()
data class HttpError(val message: String): NetworkError()
object TimeOut: NetworkError()
```

Does it generate `Prism<NetworkResult, Success>`, `Prism<NetworkResult, NetworkError>` and everything applicable to `Success`? Or does it also generate everything for `HttpError` and Prisms for `NetworkError`?

What happens if I pass arguments to `@optics`? i.e. what happens if I change it to `@optics([LENS])`? Does it fail because I cannot generate `Lens` for NetworkResult? Or does it generate a `Lens` for `Success` and doesn't complain? What happens in same scenario when `sealed class` only contains `object` and thus cannot generate any `Lens`?

We will create a ton of edge cases which I think will be extremely confusing for the users.

Reply:
Fair enough